### ISSUE TITLE
Deprecated checker is not able to detect deprecated aliases to existing functions/methods

### ISSUE THREAD

**Issue by matusvalo** [PyCQA/pylint#4618]
### Steps to reproduce
Module `threading` contains several deprecated aliases (deprecated in python 3.7 and 3.10) of existing functions:

* threading.Thread.isAlive
* threading.currentThread
* threading.activeCount
* threading.Condition.notifyAll
* threading.Event.isSet
* threading.Thread.setName
* threading.thread.getName
* threading.Thread.isDaemon
* threading.Thread.setDaemon

Current logic of detecting deprecated methods/functions does not support aliases since it is using inferred name of function for checking against list of deprecated names. Hence, even after adding the names to list of deprecated functions, pylint produces false negative result:

Given a file `a.py`:
```python
import threading

print(threading.currentThread())
```

### Current behavior

```
$ pylint test.py

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 5.00/10, +5.00)
```

### Expected behavior

```
$ pylint test.py
************* Module test
test.py:51:6: W1505: Using deprecated method currentThread() (deprecated-method)

-------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 10.00/10, -5.00)
```

### pylint --version output

Result of `pylint --version` output:

```
$ pylint --version
pylint 2.9.0-dev1
astroid 2.6.0
Python 3.8.5 (default, Jan 27 2021, 15:41:15)
[GCC 9.3.0]
```
